addon/-private/system/relationships/state/relationship.js

```diff
@@ -266,6 +266,31 @@ export default class Relationship {
     this.flushCanonicalLater();
   }
 
+  unloadInverseInternalModel(internalModel) {
+    this.unloadInverseInternalModelFromOwn(internalModel);
+    this.unloadInverseInternalModelFromInverses(internalModel);
+  }
+
+  unloadInverseInternalModelFromOwn(internalModel) {
+    this.canonicalMembers.delete(internalModel);
+    this.members.delete(internalModel);
+    this.notifyRecordRelationshipRemoved(internalModel);
+  }
+
+  unloadInverseInternalModelFromInverses(internalModel) {
+    if (!this.inverseKey) { return; }
+
+    let allMembers =
+      // we actually want a union of members and canonicalMembers
+      // they should be disjoint but currently are not due to a bug
+      this.members.toArray().concat(this.canonicalMembers.toArray());
```

Comment:
rather then making 3 arrays only to iterate them, maybe we should do something like:

```js
const unload = inverseInternalModel => {
   let relationship = inverseInternalModel._relationships.get(this.inverseKey);
   relationship.unloadInverseInternalModelFromOwn(internalModel);
}

this.members.forEach(unload);
this.canonicalMembers(unload);
```